downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -70,18 +70,16 @@ open class GitWorkingTree(
             SubmoduleWalk.forIndex(repo).use { walk ->
                 while (walk.next()) {
                     val path = "$prefix${walk.path}"
-                    paths += path
 
                     if (walk.repository == null) {
-                        logger.info {
+                        logger.warn {
                             "Git submodule at '$path' not initialized. Cannot recursively list its submodules."
                         }
-
-                        continue
-                    }
-
-                    walk.repository.use { submoduleRepo ->
-                        listSubmodules(path, submoduleRepo, paths)
+                    } else {
+                        paths += path
+                        walk.repository.use { submoduleRepo ->
```

Comment:
Meanwhile, a tiny style nit: We prefer to have empty lines before the start of code blocks, like here.

Reply:
fixed